Fix bug with renaming submodules in dynamo for new tracer

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #163580
* #163260
* #163259
* #163258
* #163137
* __->__ #163136

Differential Revision: [D82603767](https://our.internmc.facebook.com/intern/diff/D82603767)

Previously, i forgot to add handle call_module case which now will have export_root prepended to their names. Basically i want to clean up sth like:
```
graph():
      %l_self_export_root_sub_mod = call_module[target=l_self_export_root_sub_mod](%x, %y)
      %l_self_export_root_sub_mod_1 = call_module[target=l_self_export_root_sub_mod](%x, %y)
  ```
      

Dynamo graph can have call_module nodes that have messed up name due to our wrapper. 
cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @Lucaskabela

tugsbayasgalan committed:
Fix bug with renaming param/buffers in dynamo for new export

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D82603767).

tugsbayasgalan committed:
Update on "Fix bug with renaming param/buffers in dynamo for new export"


Differential Revision: [D82603767](https://our.internmc.facebook.com/intern/diff/D82603767)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Fix bug with renaming param/buffers in dynamo for new export"


Differential Revision: [D82603767](https://our.internmc.facebook.com/intern/diff/D82603767)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D82603767).

tugsbayasgalan committed:
Update on "Fix bug with renaming param/buffers in dynamo for new export"


Differential Revision: [D82603767](https://our.internmc.facebook.com/intern/diff/D82603767)

Dynamo graph can have call_module nodes that have messed up name due to our wrapper. 
cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D82603767).

avikchaudhuri commented on torch/_dynamo/functional_export.py:
```diff
@@ -118,18 +118,41 @@ def clean_name(name) -> str:
+            new_target = clean_name(old_target)
+            new_name = clean_name(node.name)
+            if new_target != old_target:
+                new_name = clean_name(node.name)
```
dupe

avikchaudhuri commented on torch/_dynamo/functional_export.py:
```diff
@@ -118,18 +118,41 @@ def clean_name(name) -> str:
+            new_name = clean_name(node.name)
+            if new_target != old_target:
+                new_name = clean_name(node.name)
+                if old_target in clean_named_module_map:
```
should pull this up before creating `new_name` or `new_target`?

avikchaudhuri reviewed (changes requested):
clean up code under `call_module`

Also, not clear from this PR what problem it is solving, please update summary

tugsbayasgalan committed:
Update on "Fix bug with renaming submodules in dynamo for new tracer"


Differential Revision: [D82603767](https://our.internmc.facebook.com/intern/diff/D82603767)

Previously, i forgot to add handle call_module case which now will have export_root prepended to their names. Basically i want to clean up sth like:
```
graph():
      %l_self_export_root_sub_mod = call_module[target=l_self_export_root_sub_mod](%x, %y)
      %l_self_export_root_sub_mod_1 = call_module[target=l_self_export_root_sub_mod](%x, %y)
  ```
      

Dynamo graph can have call_module nodes that have messed up name due to our wrapper. 
cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Fix bug with renaming submodules in dynamo for new tracer"


Differential Revision: [D82603767](https://our.internmc.facebook.com/intern/diff/D82603767)

Previously, i forgot to add handle call_module case which now will have export_root prepended to their names. Basically i want to clean up sth like:
```
graph():
      %l_self_export_root_sub_mod = call_module[target=l_self_export_root_sub_mod](%x, %y)
      %l_self_export_root_sub_mod_1 = call_module[target=l_self_export_root_sub_mod](%x, %y)
  ```
      

Dynamo graph can have call_module nodes that have messed up name due to our wrapper. 
cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]
